### PR TITLE
Rework how the json/table output formatting works

### PR DESCRIPTION
**Pull request by jasonborg**
This change reworks how the json/table output formatting works. This
Makes things a bit cleaner and will make it simpler to unit test the
command implementations.

### CODE REVIEW

File: cli/data_formatter.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    rows += values
+
+    field_buffer = '  '
+    table = ''
+
+    for i in range(len(rows)):
+      for j in range(len(rows[i])):
+        table += f'{rows[i][j]:{widths[j]}}'
+        table += field_buffer
+      table = table[:-len(field_buffer)]
```
Comment (mctavish) [inline cli/data_formatter.py:61]: field_buffer.join(fields) tends to work well for this.

File: cli/data_formatter.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      values: Array of tuples containing one row of output (1 value for each
+        column). Each tuple must have the same number of elements, which much
+        also match the length of the headers tuple.
+    """
+    widths = [0] * len(headers)
+
+    for i in range(len(headers)):
+      widths[i] = max(widths[i], len(headers[i]))
+      for j in range(len(values)):
+        widths[i] = max(widths[i], len(values[j][i]))
```
Comment (mctavish) [inline cli/data_formatter.py:45]: Every time I see a for loop in Python I assume that there's a one-liner.  Whether or not it's readable, that's another question.

```
widths = [max(len(headers[i]), max([len(v[i]) for v in values]))
          for i in range(len(headers))]
```

**Review by mctavish — APPROVED** [review]
Functionally, looks good.  A couple stylistic comments.

File: cli/data_formatter.py (hunk 1 above)
Comment (jasonborg) [reply cli/data_formatter.py]: That's much nicer, thanks!

File: cli/data_formatter.py (hunk 2 above)
Comment (jasonborg) [reply cli/data_formatter.py]: Thanks, adopted this.

**Pull request merged**